sdk/src/androidTest/java/ly/count/android/sdk/ModuleViewsTests.java

```diff
@@ -1855,6 +1855,35 @@ public void recordView_previousViewName() throws JSONException {
         validateView("test2", 0.0, 1, 2, false, true, TestUtils.map(), "_CLY_", "_CLY_", "test");
     }
 
+    /**
+     * "startView" with consent removal
+     * Validate that all running views are stopped when the view consent is removed
+     *
+     * @throws JSONException if the JSON is not valid
+     */
+    @Test
+    public void startView_consentRemoval() throws JSONException {
+        CountlyConfig countlyConfig = TestUtils.createBaseConfig();
+        countlyConfig.setRequiresConsent(true);
+        countlyConfig.setLoggingEnabled(true);
+        countlyConfig.giveAllConsents();
+        countlyConfig.setEventQueueSizeToSend(1);
+
+        Countly countly = new Countly().init(countlyConfig);
+
+        countly.views().startView("test");
+        ModuleConsentTests.validateAllConsentRequest(TestUtils.commonDeviceId, 0);
+        validateView("test", 0.0, 1, 2, true, true, TestUtils.map(), "_CLY_", "_CLY_", null);
+
+        countly.views().startView("test2");
+        validateView("test2", 0.0, 2, 3, false, true, TestUtils.map(), "_CLY_", "_CLY_", null);
+
+        countly.consent().removeConsent(new String[] { Countly.CountlyFeatureNames.views });
+        validateView("test", 0.0, 3, 6, false, false, TestUtils.map(), "_CLY_", "_CLY_", null);
+        validateView("test2", 0.0, 4, 6, false, false, TestUtils.map(), "_CLY_", "_CLY_", null);
+        ModuleConsentTests.validateConsentRequest(TestUtils.commonDeviceId, 5, new boolean[] { true, true, true, true, true, true, true, true, true, true, true, true, false, true, true });
```

Comment:
can you then give consent back and check if views start?
also can you use other ways to start a view too (like autostopped).
is there a way we can test this for automatic views? here it seems manual
if other tests are already checking those let me know

Reply:
yeap there is a way to test it for auto ones
it is kinda manual, I call callbacks manually
I will implement those too